### DISCUSSION TITLE
Writing tests

### DISCUSSION THREAD

**Discussion by pavolo99** [Project-Books/book-project#784] [Q&A]
Hi,
I would like to write some tests for my function, but the existing tests are without test data.
I don't think the tests are written effectively, because the test returns and checks the created list of books after mocking (see picture).

How should I proceed then? 
Thanks for your answer.

![image](https://user-images.githubusercontent.com/62513694/135398700-cba4560c-155f-4d28-bada-b9485282eab1.png)

**Comment by knjk04** [comment]
Hi @pavolo99,

If you haven't already, could you push this to a remote branch? Then, could you share a link to the branch? I can then clone this and take a look locally. Thanks

**Reply by pavolo99** [reply]
@knjk04, I have a problem with pushing branch because of insufficient access rights. 
Is there any problem on my side or you have to assign me the pushing rights? 
Thanks.

**Reply by knjk04** [reply]
@pavolo99 Am I right in saying that you're trying to push to this repository rather than your fork? If so, you'll need to fork the repository and then push to your fork.

See our [contributing guidelines](https://project-books.github.io/docs/development/contributing/) for the workflow

**Reply by pavolo99** [reply]
Im so sorry, I did not make a fork.

**Reply by pavolo99** [reply]
Here is link to branch: [feature/769_predefined_shelf_controller](https://github.com/pavolo99/book-project/tree/feature/769_predefined_shelf_controller)

**Reply by knjk04** [reply]
Thanks for the link.

> but the existing tests are without test data

Sorry, what do you mean by this? I thought the test data was the list of books that we're passing to the mocked call of the book service's findAll() method?
